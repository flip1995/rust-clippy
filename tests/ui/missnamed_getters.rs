#![allow(unused)]
#![warn(clippy::missnamed_getters)]

struct A {
    a: u8,
    b: u8,
}

impl A {
    fn a(&self) -> &u8 {
        &self.b
    }
}

union B {
    a: u8,
    b: u8,
}

impl B {
    unsafe fn a(&self) -> &u8 {
        &self.b
    }
}

fn main() {
    // test code goes here
}
