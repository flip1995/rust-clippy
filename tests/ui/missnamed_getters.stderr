error: getter function appears to return the wrong field
  --> $DIR/missnamed_getters.rs:11:9
   |
LL |         &self.b
   |         ^^^^^^^ help: consider using: `&self.a`
   |
   = note: `-D clippy::missnamed-getters` implied by `-D warnings`

error: getter function appears to return the wrong field
  --> $DIR/missnamed_getters.rs:22:9
   |
LL |         &self.b
   |         ^^^^^^^ help: consider using: `&self.a`

error: aborting due to 2 previous errors

